subprojects/base-services/src/main/java/org/gradle/internal/reflect/JavaReflectionUtil.java

```diff
@@ -55,4 +57,17 @@ public static <T> T newInstance(Class<T> c) {
             throw UncheckedException.throwAsUncheckedException(e);
         }
     }
+
+    /**
+     * Returns the {@link TypeToken} of a parameter specified by index.
+     *
+     * @param beanType the TypeToken of the bean
+     * @param parameterizedSuperClass the superclass
```

Comment:
:question: Maybe "superclass" was the right term to use in the original context, but for a common utility method I think this parameter should be called `parametrizedClass` instead, no?